mongo/integration/search_index_prose_test.go

```diff
@@ -311,4 +311,117 @@ func TestSearchIndexProse(t *testing.T) {
+			indexType := cursor.Current.Lookup("type").StringValue()
+			if name == indexName && queryable {
+				doc = cursor.Current
+				assert.Equal(mt, indexType, "search")
+			} else {
+				t.Logf("cursor: %s, sleep 5 seconds...", cursor.Current.String())
+				time.Sleep(5 * time.Second)
+			}
+		}
+
+		indexName = "test-search-index-case7-explicit"
+		opts = options.SearchIndexes().SetName(indexName).SetType("search")
+		index, err = view.CreateOne(ctx, mongo.SearchIndexModel{
+			Definition: definition,
+			Options:    opts,
+		})
+		require.NoError(mt, err, "failed to create index")
+		require.Equal(mt, indexName, index, "unmatched name")
+		doc = nil
+		for doc == nil {
+			cursor, err := view.List(ctx, opts)
+			require.NoError(mt, err, "failed to list")
+
+			if !cursor.Next(ctx) {
+				break
+			}
+			name := cursor.Current.Lookup("name").StringValue()
+			queryable := cursor.Current.Lookup("queryable").Boolean()
+			indexType := cursor.Current.Lookup("type").StringValue()
+			if name == indexName && queryable {
+				doc = cursor.Current
+				assert.Equal(mt, indexType, "search")
+			} else {
+				t.Logf("cursor: %s, sleep 5 seconds...", cursor.Current.String())
+				time.Sleep(5 * time.Second)
+			}
+		}
+
+		indexName = "test-search-index-case7-vector"
+		definition = bson.D{{"fields", bson.A{bson.D{{"type", "vector"}, {"path", "plot_embedding"}, {"numDimensions", 1536}, {"similarity", "euclidean"}}}}}
+		opts = options.SearchIndexes().SetName(indexName).SetType("vectorSearch")
+		index, err = view.CreateOne(ctx, mongo.SearchIndexModel{
+			Definition: definition,
+			Options:    opts,
+		})
+		require.NoError(mt, err, "failed to create index")
+		require.Equal(mt, indexName, index, "unmatched name")
+		doc = nil
+		for doc == nil {
+			cursor, err := view.List(ctx, opts)
+			require.NoError(mt, err, "failed to list")
+
+			if !cursor.Next(ctx) {
+				break
+			}
+			name := cursor.Current.Lookup("name").StringValue()
+			queryable := cursor.Current.Lookup("queryable").Boolean()
+			indexType := cursor.Current.Lookup("type").StringValue()
+			if name == indexName && queryable {
+				doc = cursor.Current
+				assert.Equal(mt, indexType, "vectorSearch")
+			} else {
+				t.Logf("cursor: %s, sleep 5 seconds...", cursor.Current.String())
+				time.Sleep(5 * time.Second)
+			}
+		}
+	})
+
+	mt.Run("case 8: Driver requires explicit type to create a vector search index", func(mt *mtest.T) {
+		ctx := context.Background()
+
+		_, err := mt.Coll.InsertOne(ctx, bson.D{})
+		require.NoError(mt, err, "failed to insert")
+
+		view := mt.Coll.SearchIndexes()
+
+		indexName := "test-search-index-case7-vector"
+		definition := bson.D{{"fields", bson.A{bson.D{{"type", "vector"}, {"path", "plot_embedding"}, {"numDimensions", 1536}, {"similarity", "euclidean"}}}}}
+		opts := options.SearchIndexes().SetName(indexName)
+		_, err = view.CreateOne(ctx, mongo.SearchIndexModel{
+			Definition: definition,
+			Options:    opts,
+		})
```

Comment:
Optional. Suggest using a struct instead of bson.D for the complex definition.
```suggestion
			type vectorDefinitionField struct {
				Type          string `bson:"type"`
				Path          string `bson:"path"`
				NumDimensions int    `bson:"numDimensions"`
				Similarity    string `bson:"similarity"`
			}

			type vectorDefinition struct {
				Fields []vectorDefinitionField `bson:"fields"`
			}

			const indexName = "test-search-index-case7-vector"
			opts := options.SearchIndexes().SetName(indexName)
			_, err = view.CreateOne(ctx, mongo.SearchIndexModel{
				Definition: vectorDefinition{
					Fields: []vectorDefinitionField{{"vector", "plot_embedding", 1536, "euclidean"}},
				},
				Options: opts,
			})
```